src/app/common/components/RouteGuard.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import { useHistory } from 'react-router';
+import ConfirmModal from '@app/common/components/ConfirmModal';
+import { RouteGuardOptions } from '@app/common/constants';
+interface IRouteGuard {
+  when: boolean;
+  title: string;
+  message: string | JSX.Element;
+  okText?: string;
+  cancelText?: string;
+}
+
+const RouteGuard: React.FunctionComponent<IRouteGuard> = ({
+  when,
+  title,
+  message,
+  okText = 'OK',
+  cancelText = 'Cancel',
+}) => {
+  const history = useHistory();
+  const [showRouteGuard, setShowRouteGuard] = React.useState(when);
+  const [currentPath, setCurrentPath] = React.useState('');
+  const unBlock = () => RouteGuardOptions.permit;
```

Comment:
Why make unBlock a function? you could just pass RouteGuardOptions.permit in the places you call it, or make it a regular variable here if you don't want to repeat that

Reply:
Good call, doesn't need to be a function :)